pulsar_adc: Add build options to include AD7944 (#1424)
analogdevicesinc/hdl: docs/projects/pulsar_adc/index.rst

Configuration modes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The Zedboard based Pulsar ADC project can be built to use either the FMC 
connector or the Pmod JA connector. The FMC_N_PMOD parameter is used to select
between the two options; 0 for PMOD and 1 for FMC. By default it is set to 1
(FMC).

The Pulsar project supports different configurations required for certain
ADCs, like the AD7944. These modes are selected using the SPI_OP_MODE parameter,
0 for normal SPI Engine connections, 1 for 3-wire "single" mode where CS drives
the SDO line while the CS line is drive by GPIO and 2 where SDO is driven by
GPIO and the CS line is driven by CS. The SPI_OP_MODE parameter must only be

> used for the FMC variant.

Author: Done

Configuration modes
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The Zedboard based Pulsar ADC project can be built to use either the FMC 
connector or the Pmod JA connector. The FMC_N_PMOD parameter is used to select
between the two options. By default it is set to 1 (FMC).

> The Pulsar project also supports more exotic configurations required for
> certain ADCs like the AD7944. These modes are selected using the SPI_OP_MODE 
> parameter.

Reviewer: ```suggestion
The Pulsar project supports different configurations required for certain
ADCs, like the AD7944. These modes are selected using the `SPI_OP_MODE` 
parameter.
```

Also, below this, I think it would be best if you could list all the possible values for the parameters and what they mean.
Otherwise one would need to enter system_project.tcl and check in the `if` statements to know the possible values.

Build examples:

Zedboard Pmod support

.. code-block::

   make FMC_N_PMOD=0

Zedboard standard configuration FMC support (default)

.. code-block::

   make FMC_N_PMOD=1 SPI_OP_MODE=0 - builds standard FMC version

Zedboard FMC support for AD7944 4-wire mode ("multi")

.. code-block::

   make FMC_N_PMOD=1 SPI_OP_MODE=1

Zedboard FMC support for AD7944 chain mode or 3-wire "single"

.. code-block::

   make FMC_N_PMOD=1 SPI_OP_MODE=2

> 

Reviewer: Build examples should be moved under the `Building the project section`, and in the `Configuration modes` just to list the possible values

Author: Done
